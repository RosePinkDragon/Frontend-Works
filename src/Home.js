import React from "react";
import { Link } from "react-router-dom";

export const HomeLink = () => (
  <Link to="/" className="fixedHome">
    Home
  </Link>
);

const Home = () => {
  return (
    <div>
      Hey those are some tasks by me
      <br />
      {/* Node tree is used to create a tree like hierarchy structure for employees */}
      <Link to="/Hierarchy" className="brand_Name color-crimson">
        Task 1: Node Tree
      </Link>
      <br />
      {/* Admin UI is a responsive UI for the admin panel */}
      <Link to="/AdminUI" className="brand_Name color-crimson">
        Task 2: Admin UI
      </Link>
      <br />
      {/* Admin UI  Class is same as Admin UI but with redux and class components */}
<<<<<<< Updated upstream
      <Link to="/AdminUI" className="brand_Name color-crimson">
=======
      <Link to="/AdminUIClass" className="brand_Name color-crimson">
>>>>>>> Stashed changes
        Task 2: Admin UI Class
      </Link>
    </div>
  );
};

export default Home;
