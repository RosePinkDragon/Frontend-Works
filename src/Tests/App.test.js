import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import Home, { HomeLink } from "../Home";

describe("App", () => {
  test("snapshot renders", () => {
    const component = renderer.create(
      <Router>
        <Home />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("HomeLink snapshot renders", () => {
    const component = renderer.create(
      <Router>
        <HomeLink />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
