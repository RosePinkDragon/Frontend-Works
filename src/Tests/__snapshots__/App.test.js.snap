// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App HomeLink snapshot renders 1`] = `
<a
  className="fixedHome"
  href="/"
  onClick={[Function]}
>
  Home
</a>
`;

exports[`App snapshot renders 1`] = `
<div>
  Hey those are some tasks by me
  <br />
  <a
    className="brand_Name color-crimson"
    href="/Hierarchy"
    onClick={[Function]}
  >
    Task 1: Node Tree
  </a>
  <br />
  <a
    className="brand_Name color-crimson"
    href="/AdminUI"
    onClick={[Function]}
  >
    Task 2: Admin UI
  </a>
  <br />
  <a
    className="brand_Name color-crimson"
    href="/AdminUIClass"
    onClick={[Function]}
  >
    Task 2: Admin UI Class
  </a>
</div>
`;
